aten/src/ATen/native/cpu/UpSampleKernel.cpp

```diff
@@ -1260,26 +1260,34 @@ void _upsample_nearest_exact1d_kernel_impl(
     output, input, false, {scales_w});
 }
 
-int _use_vectorized_kernel_cond(
+int _use_vectorized_kernel_cond_2d(
     const Tensor& output,
     const Tensor& input) {
       // This condition is used to know whether we should dispatch to a vectorized
       // kernel, or to the more general upsample_generic_Nd_kernel_impl(). For now,
       // the vectorized kernels are only optimized for channels_last and when C >= 4
       // (shape = NCHW). For a very wide range of use-cases (typically image or mask
       // resizing where we have C < 4), using upsample_generic_Nd_kernel_impl() is
-      // actually faster. On top of that, bencharmks showed that this also depends on
+      // actually faster. On top of that, benchmarks showed that this also depends on
       // the *output* size (output_H + output_W) , for both upsampling and
       // downsampling. The current 128 threshold was determined through benchmarks.
-      return ((input.is_contiguous(at::MemoryFormat::ChannelsLast)) && (input.size(-3) > 3)) || ((output.size(-2) + output.size(-1)) <= 128);
+      return ((input.is_contiguous(at::MemoryFormat::ChannelsLast)) && (input.size(1) > 3)) || ((output.size(-2) + output.size(-1)) <= 128);
 }
 
+int _use_vectorized_kernel_cond_3d(
+    // Similar to _use_vectorized_kernel_cond_2d() but for 3d resampling (e.g. videos)
+    const Tensor& output,
+    const Tensor& input) {
+      return ((input.is_contiguous(at::MemoryFormat::ChannelsLast3d)) && (input.size(1) > 3));
```

Comment:
If In this case output size does not matter, maybe, we can update above comment saying that. Otherwise, it seems like output size condition is missing...

Reply:
Good point - I can confirm that this doesn't seem to suffer from the weight pre-computation overhead for small output, unlike the 2D case. I updated the comment to reflect that.